packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.ts

```diff
@@ -68,47 +72,87 @@ export default class EthereumLedgerProvider extends LedgerProvider<HwAppEthereum
     return this.getAddresses()
   }
 
-  async signTransaction(txData: ethereum.TransactionRequest, path: string) {
-    const chainId = numberToHex((this._network as EthereumNetwork).chainId)
+  async signTransaction(txData: ethereum.EIP1559TransactionRequest | ethereum.TransactionRequest, path: string) {
+    const network = this._network as EthereumNetwork
+
+    let common
+    if (network.name !== 'local') {
+      common = EthCommon.custom(
+        {
+          name: network.name,
+          chainId: network.chainId,
+          networkId: network.networkId
+        },
+        {
+          hardfork: this._hardfork
+        }
+      )
+    }
+
+    const _txData = {
+      gasLimit: txData.gas,
+      ...txData
+    }
+
+    let tx
+
+    if (_txData.gasPrice) {
+      tx = LegacyTransaction.fromTxData(_txData, { common })
+    } else {
+      tx = FeeMarketEIP1559Transaction.fromTxData(_txData as ethereum.EIP1559TransactionRequest, { common })
+    }
+
+    const msg = tx.getMessageToSign(false)
+    const encodedMessage = _txData.gasPrice ? rlp.encode(msg) : msg
+    const encodedMessageHex = encodedMessage.toString('hex')
+
     const app = await this.getApp()
-    const tx = new EthereumJsTx({
-      ...txData,
-      chainId: hexToNumber(chainId), // HEY Could be incorrect
-      v: chainId
-    })
-    const serializedTx = tx.serialize().toString('hex')
-    const txSig = await app.signTransaction(path, serializedTx)
+    const txSig = await app.signTransaction(path, encodedMessageHex)
+
     const signedTxData = {
-      ...txData,
+      ..._txData,
       v: ensure0x(txSig.v),
       r: ensure0x(txSig.r),
       s: ensure0x(txSig.s)
     }
 
-    const signedTx = new EthereumJsTx(signedTxData)
-    const signedSerializedTx = signedTx.serialize().toString('hex')
-    return signedSerializedTx
+    let signedTx
+    if (_txData.gasPrice) {
+      signedTx = LegacyTransaction.fromTxData(signedTxData, { common })
+    } else {
+      signedTx = FeeMarketEIP1559Transaction.fromTxData(signedTxData as ethereum.EIP1559TransactionRequest, { common })
+    }
+
+    return signedTx.serialize().toString('hex')
   }
 
   async sendTransaction(options: SendOptions) {
     const addresses = await this.getAddresses()
     const address = addresses[0]
     const from = address.address
 
-    const [nonce, gasPrice] = await Promise.all([
-      this.getMethod('getTransactionCount')(remove0x(from), 'pending'),
-      options.fee ? Promise.resolve(new BigNumber(options.fee)) : this.getMethod('getGasPrice')()
-    ])
-
+    const nonce = await this.getMethod('getTransactionCount')(remove0x(from), 'pending')
     const txOptions: ethereum.UnsignedTransaction = {
       from,
       to: options.to ? addressToString(options.to) : (options.to as string),
       value: options.value,
       data: options.data,
-      gasPrice,
       nonce
     }
 
+    if (options.fee) {
```

Comment:
Wow there is a lot of duplication with JsProvider 
Definitely will be good to have a EthereumWalletProvider base class
We can do separately/later